jdbc-ucp/src/main/java/io/micronaut/configuration/jdbc/ucp/OracleUcpConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.jdbc.ucp;
+
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Nullable;
+
+import static io.micronaut.configuration.jdbc.ucp.OracleUcpConfiguration.PREFIX;
+
+/**
+ * Configuration properties for Oracle Universal Connection Pooling (UCP).
+ *
+ * @param destroyOnReload Flag indicating whether to destroy connections on reload
+ */
+@ConfigurationProperties(PREFIX)
+@Requires(property = PREFIX)
+public record OracleUcpConfiguration(@Nullable Boolean destroyOnReload) {
```

Comment:
does it make sense to combine this with https://github.com/micronaut-projects/micronaut-sql/blob/6.1.x/jdbc-ucp/src/main/java/io/micronaut/configuration/jdbc/ucp/UniversalConnectionPoolManagerConfiguration.java ?

Reply:
I thought there might be other `oracle.ucp.*` properties to set and we can use this for that purpose and looks to me it's better to keep separated from the ucp-manager.

Reply:
I rescheduled this to 4.9